src/tabmat/constructor.py

```diff
@@ -198,3 +202,35 @@ def from_csc(mat: sps.csc_matrix, threshold=0.1):
     """
     dense, sparse, dense_idx, sparse_idx = _split_sparse_and_dense_parts(mat, threshold)
     return SplitMatrix([dense, sparse], [dense_idx, sparse_idx])
+
+
+def from_formula(
+    formula: Union[str, Formula],
+    df: pd.DataFrame,
+    ensure_full_rank: bool = False,
+    context=0,
+):
+    """
+    Transform a pandas DataFrame to a SplitMatrix using a Wilkinson formula.
+
+    Parameters
+    ----------
+    formula: str
+        A formula accepted by formulaic.
+    df: pd.DataFrame
+        pandas DataFrame to be converted.
+    ensure_full_rank: bool, default False
+        If True, ensure that the matrix has full structural rank by categories.
+    """
+    if isinstance(context, int):
+        if hasattr(sys, "_getframe"):
+            frame = sys._getframe(context + 1)
+            context = LayeredMapping(frame.f_locals, frame.f_globals)
+        else:
+            context = None  # pragma: no cover
```

Comment:
I don't think we test coverage in tabmat yet, so you can remove pragma annotations,